haproxy/reload_agent.go

```diff
@@ -124,46 +123,41 @@ func (ra *ReloadAgent) handleReload(id string) {
 		ra.cache.mu.Unlock()
 	}()
 
-	response, err := ra.reloadHAProxy()
+	response, err := ra.reloadHAProxy(id)
 	if err != nil {
 		ra.cache.failReload(response)
-		log.Warning("Reload failed " + err.Error())
+		log.Warningf("Reload %s failed: %s", id, err)
```

Comment:
Maybe we could use the `WithFields` function helper here?

Would be easier also from a logging perspective to isolate logs belonging to a specific ID using the field `logger_id`, YMMV.